### ISSUE TITLE
[BUG] error creating container storage: layer not known

### ISSUE THREAD

**Issue by kanadaj** [code-ready/crc#1414]
### General information

  * OS: Fedora 31
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Baremetal-Server

## CRC version
```bash
CodeReady Containers version: 1.13.0+8070bae
OpenShift version: 4.5.1 (embedded in binary)
```
  
## CRC status
```bash
CRC VM:          Running
OpenShift:       Not Reachable
Disk Usage:      47.11GB of 140.1GB (Inside the CRC VM)
Cache Usage:     11.78GB
Cache Directory: /home/kanadaj/.crc/cache
```

## CRC config
```bash
- cpus                                  : 24
- memory                                : 20480
```

## Host Operating System
```bash
NAME=Fedora
VERSION="31 (Server Edition)"
ID=fedora
VERSION_ID=31
VERSION_CODENAME=""
PLATFORM_ID="platform:f31"
PRETTY_NAME="Fedora 31 (Server Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:31"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f31/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=31
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=31
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Server Edition"
VARIANT_ID=server
```

### Steps to reproduce

Not quite sure here; crc suddenly shut down, and trying to restart it resulted in the following error:
```bash
ERRO Error running post start: ssh command error:
command : sudo podman run  --ip 10.88.0.8 --name dnsmasq -v /var/srv/dnsmasq.conf:/etc/dnsmasq.conf -p 53:53/udp --privileged -d quay.io/crcont/dnsmasq:latest
err     : exit status 125
output  : Error: error creating container storage: layer not known
```

I'd rather not delete my crc VM since I don't want to lose the environment - I'm using CRC on my home server as a single-node Openshift cluster with persistence.

SSH-ing into the machine and retrying the command results in the same.

This is likely an issue with the upstream Openshift version: https://bugzilla.redhat.com/show_bug.cgi?id=1857224

Most likely needs an update to 4.5.3. Also, is there any way to salvage the VM?

**Comment by gbraad** [comment]
> I'd rather not delete my crc VM since I don't want to lose the environment - I'm using CRC on my home server as a single-node Openshift cluster with persistence.

CRC is not meant for this usecase as it is not setup with the best security practices. It is intended to be used as an ephemeral instance.

You clikely have to do a delete, but you can try to use `ssh` to check that the state is of the VM, as I think (due to the persistence you mention) the disk might be full. Since you have logged in before, take out what is needed ...

> Most likely needs an update to 4.5.3. Also, is there any way to salvage the VM?

the VM can NOT be updated using the regular OpenShift methods. This has been disabled as this is not wished for in a developer scenario.

**Comment by kanadaj** [comment]
> CRC is not meant for this usecase as it is not setup with the best security practices. It is intended to be used as an ephemeral instance.
I see. The issue I have is that installing OKD 4 on a single node isn't documented, and even the official documentation says it requires at least 3 masters and 2 workers. Using the sole Medium article I found would mean spinning up 7 VMs to host a cluster, which is both extremely wasteful and very limiting in terms of performance... :/

> You clikely have to do a delete, but you can try to use ssh to check that the state is of the VM, as I think (due to the persistence you mention) the disk might be full. 

No, there seems to be plenty of space left:
```bash
$ df -h
Filesystem                            Size  Used Avail Use% Mounted on
devtmpfs                              9.4G     0  9.4G   0% /dev
tmpfs                                 9.4G  168K  9.4G   1% /dev/shm
tmpfs                                 9.4G  560K  9.4G   1% /run
tmpfs                                 9.4G     0  9.4G   0% /sys/fs/cgroup
/dev/mapper/coreos-luks-root-nocrypt  131G   44G   87G  34% /sysroot
tmpfs                                 9.4G     0  9.4G   0% /tmp
/dev/vda1                             364M   88M  254M  26% /boot
/dev/vda2                             127M  6.8M  121M   6% /boot/efi
tmpfs                                 1.9G  4.0K  1.9G   1% /run/user/1000
```

Ideally I'd like to just spin up the kubelet and operators on the machine directly instead of messing around with VMs and ignition files, but sadly this is undocumented as well.

Security isn't that much of a concern though since it's only meant to be used on a private network by a single user.

That said, this issue has been reported with Openshift 4.5.2, so it might be an issue with 4.5.1 too? Is there any ETA for an updated CRC version?

**Comment by gbraad** [comment]
the release 4.5.1 and 4.5.2 are identical for only the RHCOS image having fixed a secure boot issue. 

A new version will be available perhaps soon. Perhaps this is resolved? However, I am not convinced what causes this...

**Comment by kanadaj** [comment]
@gbraad I think it was fixed in 4.5.3, some fsync related issue. The bug report was https://bugzilla.redhat.com/show_bug.cgi?id=1857224 and the fix is in https://access.redhat.com/errata/RHBA-2020:2956, fairly sure that's the one

**Comment by gbraad** [comment]
This might be released as CRC soon-ish... in about 1 1/2 to 2 weeks

**Comment by praveenkumar** [comment]
@kanadaj Does this fix with crc-1.14.0 since it have 4.5.3 version?

**Comment by kanadaj** [comment]
@praveenkumar I haven't tried yet, I ended up setting up my homelab with Rancher instead in the meantime, but I believe it should be fixed with 1.14.0. It would have been better if I could use crc for my homelab, but single node setups aren't much of a thing outside of CRC (running 3-5 VMs on a host is not acceptable imo) and I needed something up then.

**Comment by praveenkumar** [comment]
@kanadaj Ack, I am closing this issue in that case.
